pulsar/producer_impl.go

```diff
@@ -190,8 +190,10 @@ func (p *producer) internalCreatePartitionsProducers() error {
 	p.producers = make([]Producer, newNumPartitions)
 
 	// Copy over the existing consumer instances
-	for i := 0; i < oldNumPartitions; i++ {
-		p.producers[i] = oldProducers[i]
+	if oldProducers != nil {
```

Comment:
This could be moved few lines above, where there is already a block that checks for != nil

Reply:
Cool, updated! PTAL again